File: src/main/java/de/com/fdm/bot/commands/ListChannelsCommand.java
package de.com.fdm.bot.commands;

import de.com.fdm.bot.Command;
import de.com.fdm.bot.Parameters;
import de.com.fdm.twitch.tmi.TmiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ListChannelsCommand implements Command {
    private final TmiService tmiService;

    public ListChannelsCommand(@Autowired TmiService tmiService) {
        this.tmiService = tmiService;
    }

    @Override
    public String execute(Parameters params) {
        return tmiService.getChannels().toString();
    }
}

File: src/main/java/de/com/fdm/bot/commands/ListChannelsCommand.kt
package de.com.fdm.bot.commands

import de.com.fdm.bot.Command
import de.com.fdm.bot.api.twitch.tmi.TmiMessage
import de.com.fdm.bot.api.twitch.tmi.TmiReceiverService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class ListChannelsCommand(@param:Autowired private val tmiReceiverService: TmiReceiverService) : Command {
    override fun execute(tmiMessage: TmiMessage): String {
        return tmiReceiverService.getChannels().toString()
    }
}